EventHandlers/oncommanderror.py

```diff
@@ -1,26 +1,36 @@
+from typing import Any
+
 from discord.ext import commands
+from Translations.Translations import get_text as translate
+from Helpers.TranslationHelper import get_culture_from_context as culture
+from Helpers.log import error as log_error
+
 
 class OnCommandError(commands.Cog, name="on_command_error"):
-    def __init__(self, bot):
+    def __init__(self, bot: commands.Bot):
         self.bot = bot
 
     @commands.Cog.listener(name="on_command_error")
-    async def on_command_error(self, ctx, error):
-        print(error)
+    async def on_command_error(self, ctx: commands.Context, error: Any):
+        """
+        Handles any errors thrown by the commands.
+        :param ctx: The current context. (discord.ext.commands.Context)
+        :param error: The current error. (Any)
+        """
 
+        # Log the error
+        log_error(error)
+
+        # Notify user for MissingRequiredArgument errors
         if isinstance(error, commands.MissingRequiredArgument):
             command_name = ctx.message.content.split(" ")[0]
-            msg = (
-                "My command `"
-                + command_name
-                + "` requires more arguments. `"
-                + error.param.name
-                + "` is missing."
-            )
-        else:
-            msg = "Unrecognized command"
+            msg = translate("err_missing_parameter", await culture(ctx)).format(command_name, error.param.name)
```

Comment:
not a fan of translating the parameters since these have a tendency to be renamed or have duplicates.
I do like the idea though, and we might show the help text for the command

Reply:
On second thought, translating the parameters is not desired. But then I would indeed show the help text. Because if the bot and the help are set to Dutch and then the bot suddenly tells you that you are missing the argument `mention` or something, it might get a bit confusing and it defeats the purpose of the translations a bit imo

Reply:
issue logged: https://github.com/nerdland-unofficial-fans/nerdlandbot/issues/49